src/Raven.Embedded/EmbeddedServer.cs

```diff
@@ -192,7 +205,7 @@ public async Task<Uri> GetServerUriAsync(CancellationToken token = default)
             return (await server.Value.WithCancellation(token).ConfigureAwait(false)).ServerUrl;
         }
 
-        private void ShutdownServerProcess(Process process)
+        private void ShutdownServerProcess(Process process, CancellationToken cancellationToken) // Solution B
```

Comment:
I think that I would rather just keep the token that we already got in the `Start` method.
This simplify everything.
But that means that if we can't really cancel that, since we cancel that, we will call `Dispose`, then it is `Kill` since the token is already cancelled.

So you need two stages here anyway. One to _initiate_ shutdown and for _graceful_ shutdown.

Having said that, `Kill` isn't instant either, so that can cause you to block here for a while too.

Reply:
@ayende Yes, the approach with the token via Start is only a STOP token, and would not support stop timeout via cancellation.

We would prefer method 2 stop, that also supports cancellation.

Do you agree? We could focus on that in this PR.